cms/cms_toolbars.py

```diff
@@ -313,13 +313,16 @@ def add_logout_button(self, parent):
     def add_language_menu(self):
         if settings.USE_I18N and not self._language_menu:
             self._language_menu = self.toolbar.get_or_create_menu(LANGUAGE_MENU_IDENTIFIER, _('Language'), position=-1)
-            language_changer = getattr(self.request, '_language_changer', DefaultLanguageChanger(self.request))
+            page = self.request.current_page
             for code, name in get_language_tuple(self.current_site.pk):
                 try:
-                    url = language_changer(code)
-                except NoReverseMatch:
+                    content = page.get_title_obj(code, fallback=False)
+                    url = get_object_preview_url(content)
+                    disabled = False
```

Comment:
Why has the disabled item been added? I do remember seeing the language menu disabling before, surely that's the active?

Reply:
active and disabled are two different things. Active means that it's the item that is currently selected. Disabled means that you can't click on the item to begin with.